internal/request/graphql/parser/request.go

```diff
@@ -23,69 +23,71 @@ import (
 
 // ParseRequest parses a root ast.Document, and returns a formatted Request object.
 // Requires a non-nil doc, will error otherwise.
-func ParseRequest(schema gql.Schema, doc *ast.Document) (*request.Request, []error) {
+func ParseRequest(schema gql.Schema, doc *ast.Document, options *client.GQLOptions) (*request.Request, []error) {
 	if doc == nil {
 		return nil, []error{client.NewErrUninitializeProperty("ParseRequest", "doc")}
 	}
+	exe, err := gql.BuildExecutionContext(gql.BuildExecutionCtxParams{
+		Schema:        schema,
+		AST:           doc,
+		OperationName: options.OperationName,
+		Args:          options.Variables,
+	})
+	if err != nil {
+		return nil, []error{err}
+	}
 
 	r := &request.Request{
 		Queries:      make([]*request.OperationDefinition, 0),
 		Mutations:    make([]*request.OperationDefinition, 0),
 		Subscription: make([]*request.OperationDefinition, 0),
 	}
```

Comment:
thought (out of scope): Since we are now following the specs and explicitly allowing only one operation even if multiple are defined, it probably makes sense to refactor the `request.Request` type at some point.

Reply:
The spec allows multiple, but if multiple operations are defined, technically you're supposed to specify which named op to run within in the document

Reply:
> 6.1Executing Requests
> To execute a request, the executor must have a parsed Document and a selected operation name to run if the document defines multiple operations, otherwise the document is expected to only contain a single operation.

Reply:
Additionally, I would like to support multiple operations in the future, to support something akin to sub queries.

So you can use the result of one query into the doc set of another query.

Reply:
> The spec allows multiple, but if multiple operations are defined, technically you're supposed to specify which named op to run within in the document

My thought still stands here as although the gql can have multiple ops, the execution will only use one. 

> Additionally, I would like to support multiple operations in the future, to support something akin to sub queries.

We'll have to think carefully about the design of this one and that might also impact what `request.Request` looks like.

Reply:
> Additionally, I would like to support multiple operations in the future, to support something akin to sub queries.
> 
> So you can use the result of one query into the doc set of another query.

Can this be achieved with fragments?

Reply:
> > Additionally, I would like to support multiple operations in the future, to support something akin to sub queries.
> > 
> > So you can use the result of one query into the doc set of another query.
> 
> Can this be achieved with fragments?

I don't see how off the top of my head, but open to seeing what if anything you have in mind.